src/precompiles/mod.rs

```diff
@@ -168,20 +156,50 @@ pub fn berlin_precompiles(
         None => return Some(PrecompileResult::Err(ExitError::OutOfGas)),
     };
 
-    match address.to_low_u64_be() {
-        1 => Some(ECRecover::run(input, target_gas, context)),
-        2 => Some(SHA256::run(input, target_gas, context)),
-        3 => Some(RIPEMD160::run(input, target_gas, context)),
-        4 => Some(Identity::run(input, target_gas, context)),
-        5 => Some(ModExp::<Berlin>::run(input, target_gas, context)), // TODO gas changes
-        6 => Some(BN128Add::<Istanbul>::run(input, target_gas, context)),
-        7 => Some(BN128Mul::<Istanbul>::run(input, target_gas, context)),
-        8 => Some(BN128Pair::<Istanbul>::run(input, target_gas, context)),
-        9 => Some(Blake2F::run(input, target_gas, context)),
+    match address.0 {
+        ECRecover::ADDRESS => Some(ECRecover::run(input, target_gas, context)),
+        SHA256::ADDRESS => Some(SHA256::run(input, target_gas, context)),
+        RIPEMD160::ADDRESS => Some(RIPEMD160::run(input, target_gas, context)),
+        Identity::ADDRESS => Some(Identity::run(input, target_gas, context)),
+        modexp::ADDRESS => Some(ModExp::<Berlin>::run(input, target_gas, context)), // TODO gas changes
+        bn128::addresses::ADD => Some(BN128Add::<Istanbul>::run(input, target_gas, context)),
+        bn128::addresses::MUL => Some(BN128Mul::<Istanbul>::run(input, target_gas, context)),
+        bn128::addresses::PAIR => Some(BN128Pair::<Istanbul>::run(input, target_gas, context)),
+        Blake2F::ADDRESS => Some(Blake2F::run(input, target_gas, context)),
         #[cfg(feature = "contract")]
-        EXIT_TO_NEAR_ID => Some(ExitToNear::run(input, target_gas, context)),
+        ExitToNear::ADDRESS => Some(ExitToNear::run(input, target_gas, context)),
         #[cfg(feature = "contract")]
-        EXIT_TO_ETHEREUM_ID => Some(ExitToEthereum::run(input, target_gas, context)),
+        ExitToEthereum::ADDRESS => Some(ExitToEthereum::run(input, target_gas, context)),
         _ => None,
     }
 }
+
+/// const fn for making an address by concatenating the bytes from two given numbers,
+/// Note that 32 + 128 = 160 = 20 bytes (the length of an address). This function is used
+/// as a convenience for specifying the addresses of the various precompiles.
+const fn make_address(x: u32, y: u128) -> [u8; 20] {
+    let x_bytes = x.to_be_bytes();
+    let y_bytes = y.to_be_bytes();
+    [
+        x_bytes[0],
+        x_bytes[1],
+        x_bytes[2],
+        x_bytes[3],
+        y_bytes[0],
+        y_bytes[1],
+        y_bytes[2],
+        y_bytes[3],
+        y_bytes[4],
+        y_bytes[5],
+        y_bytes[6],
+        y_bytes[7],
+        y_bytes[8],
+        y_bytes[9],
+        y_bytes[10],
+        y_bytes[11],
+        y_bytes[12],
+        y_bytes[13],
+        y_bytes[14],
+        y_bytes[15],
+    ]
+}
```

Comment:
This is quite simple, but I'd prefer if we have unit-tests for at least 0..9 addresses to make sure that we never ever accidentally break it. Otherwise, that would be hard to debug in the future.

Reply:
The function was being implicitly tested by the tests checking the `ExitToNear` and `ExitToEthereum` addresses, as well as the standard precompiles sanity test (since that relies on dispatching to the right addresses).

That said, you are right that more tests are always a good thing. I have added some tests specifically for this function in https://github.com/aurora-is-near/aurora-engine/pull/107/commits/fe37d3ba83270aa26541d1d8c9d7ce63b7a438c4

Reply:
@birchmd got it, thanks for adding this. LGTM!